instance/models/openedx_instance.py

```diff
@@ -127,14 +128,22 @@ def database_name(self):
         """
         The database name used for external databases/storages, if any.
         """
-        name = self.internal_lms_domain.replace('.', '_')
-        # Escape all non-ascii characters and truncate to 50 chars.
-        # The maximum length for the name of a MySQL database is 64 characters.
-        # But since we add suffixes to database_name to generate unique database names
-        # for different services (e.g. xqueue) we don't want to use the maximum length here.
-        allowed = string.ascii_letters + string.digits + '_'
-        escaped = ''.join(char for char in name if char in allowed)
-        return truncate_name(escaped, length=50)
+        current_database_name = self.current_database_name
+
+        if current_database_name is not None and current_database_name == '':
+            name = self.internal_lms_domain.replace('.', '_')
+            # Escape all non-ascii characters and truncate to 50 chars.
+            # The maximum length for the name of a MySQL database is 64 characters.
+            # But since we add suffixes to database_name to generate unique database names
+            # for different services (e.g. xqueue) we don't want to use the maximum length here.
+            allowed = string.ascii_letters + string.digits + '_'
+            escaped = ''.join(char for char in name if char in allowed)
+            current_database_name = truncate_name(escaped, length=40)
```

Comment:
We'll have to do the `length=50` -> `length=40` change in a separate PR, so that we can first set the new `database_name` field to the current value (capped at 50 chars) for the existing instances.